### PR TITLE
Topics

### PR DESCRIPTION
**Pull request by ojcorrigan**
Have got it so you can click on a comment and it takes you to a dedicated page for that topic that you can send to someone else, also have it so if you search for an invalid topic you get a not found page.

### CODE REVIEW

File: src/components/Article-comments.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  err,
+  setCommentChange,
+  username,
+  article_id,
+}) => {
+  useEffect(() => {
+    setCommentChange(false);
+    getComments(article_id)
+      .then((data) => {
+        function compare(a, b) {
```
Comment (karlaferry) [inline src/components/Article-comments.jsx:20]: Extract this helper function and put it in a separate js file please :) It's always best practice to have a dedicated js file for helper functions.

File: src/components/Article-votes.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      <button
+        disabled={voted}
+        onClick={() => {
+          incVote(-1, article_id);
+          setVoteState(-1);
+          setVoted(true);
+        }}
+      >
+        Down Vote
+      </button>
```
Comment (karlaferry) [inline src/components/Article-votes.jsx:29]: My only comment about the voting buttons is when I click either upvote or downvote, they both get disabled. What if a user has voted and wants to change their mind?

File: src/components/Articles.jsx [hunk 3]
```diff
@@ -11,13 +13,26 @@ const Articles = () => {
...
+      .then((data) => {
+        setArticles(data.articles);
+      })
+      .catch((error) => {
+        setErr(error);
+      });
   }, [sortBy, order, seeOnly]);
 
+  if (err) {
+    return <RouteMissing></RouteMissing>;
```
Comment (karlaferry) [inline src/components/Articles.jsx:33]: Components should look like `<RouteMissing />`

**Review by karlaferry — CHANGES_REQUESTED** [review]
Good job Owen. I've left a couple of comments here on github and I have some additional ones as I previewed your app:

- A user would have a smoother sorting experience if it's clear that the criteria (author, topic, date) and order (asc, desc) are two separate things (ie. you can sort by author asc or desc). Right now it's unclear, but maybe a simple label might make it clearer.
- I appreciate the fact that you thought of error handling right now, but I would recommend setting up a Route path in your App.js for `path="/*"`. This means if a user tries to access any other paths that you haven't defined, your app takes them to the 404 page. Better yet, once you've gone deeper into error handling, you could make an error component (instead of just a route missing component) that will handle all of the errors for you.

**Review by jadelandeg — CHANGES_REQUESTED** [review]
When I click on the topics I get your route not found page (very nice dog) and then when I try to click back to home I can't. Just get stuck on the error page.

Also get the error page if I try to sort by vote (not sure if you've done that yet)

**Pull request closed without merging**